modules/theme-tools/responsive-videos.php

```diff
@@ -42,3 +43,51 @@ function jetpack_responsive_videos_embed_html( $html ) {
 
 	return '<div class="jetpack-video-wrapper">' . $html . '</div>';
 }
+
+/**
+ * Check if oEmbed is YouTube or Vimeo before wrapping.
+ *
+ * @return string
+ */
+function jetpack_responsive_videos_maybe_wrap_oembed( $html, $url ) {
+	if ( empty( $html ) || ! is_string( $html ) || ! $url ) {
+		return $html;
+	}
+
+	$is_video = false;
+	/**
+	 * oEmbed Video Providers.
+	 *
+	 * A whitelist of oEmbed video provider Regex patterns to check against before wrapping the output.
+	 *
+	 * @since 3.7
+	 *
+	 * @param array $video_patterns oEmbed video provider Regex patterns.
+	 */
+	$video_patterns = apply_filters( 'jetpack_responsive_videos_oembed_videos', array(
+		'https?://((m|www)\.)?youtube\.com/watch',
+		'https?://((m|www)\.)?youtube\.com/playlist',
+		'https?://youtu\.be/',
+		'https?://(.+\.)?vimeo\.com/',
+		'https?://(www\.)?dailymotion\.com/',
+		'https?://dai.ly/',
+		'https?://(www\.)?hulu\.com/watch/',
+		'https?://wordpress.tv/',
+		'https?://(www\.)?funnyordie\.com/videos/',
+		'https?://vine.co/v/',
+		'https?://(www\.)?collegehumor\.com/video/',
+		'https?://(www\.|embed\.)?ted\.com/talks/'
+	) );
+
+	// Merge patterns to run in a single preg_match call.
+	$video_patterns = '(' . implode( '|', $video_patterns ) . ')';
+
+	$is_video = preg_match( $video_patterns, $url );
+
+	// If the oEmbed is a video, wrap it in the responsive wrapper.
+	if ( $is_video ) {
```

Comment:
php.net says to always compare preg_match bools with `===` comparisons because of <a href="http://php.net/manual/en/function.preg-match.php#refsect1-function.preg-match-returnvalues">reasons</a>